test/e2e/tests/metrics/developer-options-sentry.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { Suite } from 'mocha';
+import { withFixtures } from '../../helpers';
+import FixtureBuilder from '../../fixture-builder';
+import { Driver } from '../../webdriver/driver';
+import { loginWithBalanceValidation } from '../../page-objects/flows/login.flow';
+import SettingsPage from '../../page-objects/pages/settings-page';
+import HeaderNavbar from '../../page-objects/pages/header-navbar';
+import DevelopOptions from '../../page-objects/pages/deveoper-options';
+import ErrorPage from '../../page-objects/pages/ErrorPage';
+
+const triggerCrash = async (driver: Driver): Promise<void> => {
+  const headerNavbar = new HeaderNavbar(driver);
+  await headerNavbar.openSettingsPage();
+  const settingsPage = new SettingsPage(driver);
+  await settingsPage.check_pageIsLoaded();
+  await settingsPage.goToDevelopOptionSettings();
+
+  const developOptionsPage = new DevelopOptions(driver);
+  await developOptionsPage.check_pageIsLoaded();
+  await developOptionsPage.clickGenerateCrashButton();
+};
+
+describe('Developer Options - Sentry', function (this: Suite) {
+  it('gives option to cause a page crash and provides sentry form to report', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        title: this.test?.fullTitle(),
+        ignoredConsoleErrors: ['ignore-all'],
+      },
+      async ({ driver }: { driver: Driver }) => {
+        await loginWithBalanceValidation(driver);
+        await triggerCrash(driver);
+        const errorPage = new ErrorPage(driver);
+        await errorPage.check_pageIsLoaded();
+        await errorPage.validate_errorMessage();
+        await errorPage.submitToSentryUserFeedbackForm();
+      },
+    );
+  });
+
+  it('gives option to cause a page crash and offer contact support option', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        title: this.test?.fullTitle(),
+        ignoredConsoleErrors: ['ignore-all'],
+      },
+      async ({ driver }: { driver: Driver }) => {
+        await loginWithBalanceValidation(driver);
+        await triggerCrash(driver);
+
+        const errorPage = new ErrorPage(driver);
+        await errorPage.check_pageIsLoaded();
+
+        await errorPage.contactAndValidateMetamaskSupport();
+      },
+    );
+  });
+});
```

Comment:
could we add another spec to test the flow `Describe What Happened`?

For that, we would need to mock a Sentry request with the expected payload, (which should include our description?), and then check that this request happened after clicking Submit?

There are similar examples in `errors.spec` file. We would need to use the `getSeenRequests()` method in the test, to verify the request happened with the expected body.
What do you think? Happy to help too :)
![Screenshot from 2024-10-17 12-27-08](https://github.com/user-attachments/assets/d27a24ea-16fc-494d-b0ae-f1d1d3a103ee)

Reply:
Thanks for suggestion, this is something i've been struggling with, as sentry-form is something outside the DOM.

> could we add another spec to test the flow Describe What Happened? For that, we would need to mock a Sentry request with the expected payload, (which should include our description?), and then check that this request happened after clicking Submit?

Actually the first test is about sentry form, => `gives option to cause a page crash and provides sentry form to report`, while we stopped at `await this.driver.waitForSelector(this.sentryReportForm);`.  this is because this form is outside `.app-content` DOM and `sentry-feedback` was not able to be selected. You can inspect more in the following screenshot of DOM.


<img width="513" alt="截屏2024-10-17 23 29 57" src="https://github.com/user-attachments/assets/82366e83-60f5-42a0-ab4c-6e0078accd54">

=> which means we cannot find the element 
```
<failure message="Waiting for element to be located By(xpath, .//btn//btn--primary[(contains(string(.), &apos;Submit&apos;) or contains(string(.), &apos;Submit&apos;))])
Wait timed out after 10062ms" type="TimeoutError"><![CDATA[TimeoutError: Waiting for element to be located By(xpath, .//btn//btn--primary[(contains(string(.), 'Submit') or contains(string(.), 'Submit'))])
Wait timed out after 10062ms
```

i would be happy to get deeper with this test but unfortunately even in UI side they were difficult to select :(

Reply:
oh I see what you mean! Thank you for your explanation :heart:  it make sense to leave out this case then :+1:

Reply:
@seaona I changed the implementation of sentry form by using a different API, now it is using customized form and we can completely control and test the behavior 🥳 implementation [here](https://github.com/MetaMask/metamask-extension/pull/27857/commits/15be59ce4601178d8ce07bb4bf696799f94f455d)

Reply:
<img width="645" alt="截屏2024-10-25 02 05 56" src="https://github.com/user-attachments/assets/92b914df-debe-4575-9e42-30e8537db242">

Reply:
There are some issues when running it in CI and will investigate in another ticket: https://github.com/MetaMask/MetaMask-planning/issues/3613